symbol-table/src/types/functions/function.rs

```diff
@@ -127,7 +127,9 @@ impl FunctionType {
     pub fn num_inputs(&self) -> usize {
         self.inputs
             .iter()
-            .fold(0, |acc, function_input| acc + function_input.count())
+            .filter(|function_input| !function_input.is_self())
```

Comment:
I seem to have missed it before, but `filter_self_inputs` could be used here to reduplicate the code a bit.